editor/editor_help.cpp

```diff
@@ -3320,142 +3370,464 @@ String EditorHelpBit::get_theme_item_description(const StringName &p_class_name,
+			name = name.trim_prefix("metadata/");
+		} else if (class_name == "ProjectSettings") {
+			visible_type = TTR("Project Setting:");
+		} else if (class_name == "EditorSettings") {
+			visible_type = TTR("Editor Setting:");
+		} else {
+			visible_type = TTR("Property:");
+		}
+		help_data = _get_property_help_data(class_name, item_name);
+	} else if (item_type == "internal_property") {
+		visible_type = TTR("Internal Property:");
+		help_data = HelpData();
+		help_data.description = "[color=$EditorHelpBitCommentColor][i]" + TTR("This property can only be set in the Inspector.") + "[/i][/color]";
+	} else if (item_type == "method") {
+		visible_type = TTR("Method:");
+		help_data = _get_method_help_data(class_name, item_name);
+	} else if (item_type == "signal") {
+		visible_type = TTR("Signal:");
+		help_data = _get_signal_help_data(class_name, item_name);
+	} else if (item_type == "theme_item") {
+		visible_type = TTR("Theme Property:");
+		help_data = _get_theme_item_help_data(class_name, item_name);
+	} else {
+		ERR_FAIL_MSG("Invalid tooltip type '" + item_type + "'. Valid types are 'class', 'property', 'internal_property', 'method', 'signal', and 'theme_item'.");
+	}
 
-	doc_class_name = class_name;
+	symbol_class_name = class_name;
+	symbol_type = item_type;
+	symbol_visible_type = visible_type;
+	symbol_name = name;
 
-	String formatted_text;
+	if (help_data.description.is_empty()) {
+		help_data.description = "[color=$EditorHelpBitCommentColor][i]" + TTR("No description available.") + "[/i][/color]";
+	}
 
-	// Exclude internal properties, they are not documented.
-	if (type == "internal_property") {
-		formatted_text = "[i]" + TTR("This property can only be set in the Inspector.") + "[/i]";
-		set_text(formatted_text);
-		return;
+	dirty = true;
+	queue_redraw();
+}
+
+void EditorHelpBit::set_custom_text(const String &p_type, const String &p_name, const String &p_description) {
+	symbol_class_name = String();
+	symbol_type = String();
+	symbol_visible_type = p_type;
+	symbol_name = p_name;
+
+	help_data = HelpData();
+	help_data.description = p_description;
+
+	dirty = true;
+	queue_redraw();
+}
+
+void EditorHelpBit::prepend_description(const String &p_text) {
+	if (help_data.description.is_empty()) {
+		help_data.description = p_text;
+	} else {
+		help_data.description = p_text + "\n\n" + help_data.description;
 	}
 
-	String title;
-	String description;
+	dirty = true;
+	queue_redraw();
+}
+
+void EditorHelpBit::set_content_height_limits(float p_min, float p_max) {
+	ERR_FAIL_COND(p_min > p_max);
+
+	content_min_height = p_min;
+	content_max_height = p_max;
+	content->set_custom_minimum_size(Size2(512 * EDSCALE, content_min_height));
+
+	dirty = true;
+	queue_redraw();
+}
+
+EditorHelpBit::EditorHelpBit(const String &p_symbol) {
+	add_theme_constant_override("separation", 0);
+
+	title = memnew(RichTextLabel);
+	title->set_fit_content(true);
+	title->set_selection_enabled(true);
+	//title->set_context_menu_enabled(true);
```

Comment:
Commented-out code should be removed.

Reply:
Added TODO comment.

Reply:
Why would you want the context menu though?

Reply:
* See #74114.

Some people like me are used to copy text with context menu in certain scenarios. I have no idea why I'm not using <kbd>Ctrl + C</kbd> here.